File: Projeto_5-KW/src/Professor.cpp
#include "Professor.h"
#include <iostream>

Professor::Professor() : Usuario(), areaDeAtuacao("") {}

Professor::Professor(const std::string& nome, const std::string& email, const std::string& area)
    : Usuario(nome, email, "Professor"), areaDeAtuacao(area) {}

void Professor::adicionarDisciplina(const std::string& disciplina) {
    disciplinasMinistradas.push_back(disciplina);
}

void Professor::gerarRelatorio() const {
    std::cout << "Relatório do Professor:\n";
    std::cout << "Nome: " << nome << "\nÁrea de Atuação: " << areaDeAtuacao << "\n";
    std::cout << "Disciplinas ministradas:\n";
    for (const auto& disc : disciplinasMinistradas) {
        std::cout << "- " << disc << "\n";
    }
}

File: isaaaaaac5/professor.py
class Usuario:
    def __init__(self, nome="", email="", tipo=""):
        self.nome = nome
        self.email = email
        self.tipo = tipo
class Professor(Usuario):
    def __init__(self, nome="", email="", area_de_atuacao=""):
        super().__init__(nome, email, "Professor")
        self.area = area_de_atuacao
        self.disciplinas_ministradas = []

    def adicionar_disciplina(self, disciplina):
        self.disciplinas_ministradas.append(disciplina)

    def gerar_relatorio(self):
        print("Relatório do Professor:")
        print(f"Nome: {self.nome}")
        print(f"Área de Atuação: {self.area}")
        print("Disciplinas ministradas:")
        for disc in self.disciplinas_ministradas:
            print(f"- {disc}")
